centreon-open-tickets/www/modules/centreon-open-tickets/class/rule.php

```diff
@@ -160,61 +160,99 @@ public function getMacroNames($rule_id, $widget_id)
         return $result;
     }
 
-    public function loadSelection($db_storage, $cmd, $selection)
+    /**
+     * @param CentreonDB|null $dbStorage
+     * @param string $cmd
+     * @param string $selection
+     *
+     * @return array<string, array>
+     */
+    public function loadSelection(?CentreonDB $dbStorage, string $cmd, string $selection): array
```

Comment:
The typing of $cmd and $selection is ricky because sometimes this method is called with arguments that have not been checked before. No ?

Reply:
Seen together it's ok